### PR TITLE
client: rename follower handle var

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #7431

### What is changed and how does it work?
We need to clearly define the range of APIs that a follower handle can handle

```commit-message
```

### Check List

Tests

- Unit test

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -707,12 +707,12 @@ func (c *client) UpdateOption(option DynamicOption, value interface{}) error {
 			return errors.New("[pd] invalid value type for EnableTSOFollowerProxy option, it should be bool")
 		}
 		c.option.setEnableTSOFollowerProxy(enable)
-	case EnableFollowerHandle:
+	case EnableFollowerHandleRegionRequest:
```
Comment (rleungx) [inline client/client.go:710]: If we have more and more requests, will the option become bloated in the future?

Comment (CabinfeverB) [reply client/client.go:710]: It'll be more, but I don't think it'll be bloated. If there are more options, we can provide a function to turn all options on or off, but we must be able to control them separately

**Review by lhy1024 — APPROVED** [review]

**Comment by CabinfeverB** [comment]
After discussion, because there is only region api at present, and the variables of pd client are generally not directly exposed to users, we can not change it for the time being

**Pull request closed without merging**